ethereal-dex-tri -> lib.rs -> AUXILIARY adds

alanci17 opened the pull request.

alanci17 committed:
AUXILIARY adds 

left undone REAL/EUXLP check in 'spot_price' fn

alanci17 committed:
AUXILIARY adds

left undone REAL/EUXLP check in 'spot_price' fn

alanci17 committed:
AUX adds. added fee & modified operator 

added fee on 'sim swap' method
modified operator on 'in_given_price' method

alanci17 committed:
Delta adds: tokens whitelist, from treasury to aa treasury methods

1.0_Whitelisted tlp, exrd, xrd, eusd, real, euxlp at instantiation
      (kinda raw tbf, it is what it is) 
1.1_Modified 'deposit' method to avoid missing funds due to 
      potential Vaults overwriting     
2.0_Added from treasury to aa treasury methods:   
      't_to_aat_some'  to trans some or trans all,
      't_to_aat_all'    to trans all,
      both restricted to alpha

zygomeb commented on contracts/ethereal-delta/src/lib.rs:
```diff
@@ -83,12 +93,13 @@ mod delta {
     }
 
-    pub fn deposit(&mut self, input: Bucket) {
+    pub fn deposit(&mut self, input: Bucket) -> Option<Bucket> {
```
changing this would break Alpha's assumptions lol 
in general try not to change the behavior/signatures without verifying how the entire system depends on them

i.e. this function is meant to never fail and make a new resource vault if it doesn't exist

zygomeb commented on contracts/ethereal-delta/src/lib.rs:
```diff
@@ -139,6 +150,32 @@ mod delta {
 
-    // todo move from treasury to aa treasury
+    // from treasury to aa treasury, take some or take all
+    pub fn t_to_aat_some(&mut self, a_real: Option<Decimal>, a_euxlp: Option<Decimal>) {
```
Yes this is a nice addition thanks for filling it out

alanci17 commented:
So, maybe i'm wrong :), being technically a globalized component, a random
user/attacker could call the method and stake whatever asset he wants
within KeyValueStore?

On Sat, 7 Oct 2023, 21:52 zygomeb, ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In contracts/ethereal-delta/src/lib.rs
> <https://github.com/EtherealDAO/ethereal-contracts/pull/1#discussion_r1349567261>
> :
>
> > @@ -83,12 +93,13 @@ mod delta {
>        self.power_delta.take_all()
>      }
>
> -    pub fn deposit(&mut self, input: Bucket) {
> +    pub fn deposit(&mut self, input: Bucket) -> Option<Bucket> {
>
> changing this would break Alpha's assumptions lol
> in general try not to change the behavior/signatures without verifying how
> the entire system depends on them
>
> i.e. this function is meant to never fail and make a new resource vault if
> it doesn't exist
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/EtherealDAO/ethereal-contracts/pull/1#pullrequestreview-1663103416>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXBFRQRB3N2FBZM2UIZO72LX6GXGFAVCNFSM6AAAAAA5N23NPKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTMNRTGEYDGNBRGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

zygomeb commented:
@alanci17 the deposit method is, well, a way to deposit into the treasury which itself is glad to take donations from just about anyone of any resource lol

alanci17 commented:
Got it :). So, if i may ask, why the need to whitelist those specific
assets?

On Sat, 7 Oct 2023, 22:10 zygomeb, ***@***.***> wrote:

> @alanci17 <https://github.com/alanci17> the deposit method is, well, a
> way to deposit into the treasury which itself is glad to take donations
> from just about anyone of any resource lol
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/EtherealDAO/ethereal-contracts/pull/1#issuecomment-1751808180>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXBFRQRYZBIYMUX3OT6LRU3X6GZNHAVCNFSM6AAAAAA5N23NPKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTONJRHAYDQMJYGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

zygomeb commented:
@alanci17 there's not really a whitelist (there used to be one in prior versions). the only reason why there's a mapping via KVS is for it to scale infinitely

alanci17 committed:
erased whitelist. restored 'deposit' method

1_erased assets whitelisting process within instantiation function
2_restored original 'deposit' method (added semicolon on closure)

alanci17 committed:
Aux adds, 'add_liquidity'/'internal_swap' math operators execution order revisited 

1_AUXILIARY (for interop)
  1.0_Completed 'in_given_out' method, formula derived from 
        'internal_swap' method, possible negative/NaNs outcomes?
  1.1_Completed 'sim_swap' method, replicated formula from 
        'internal_swap' method

2_Math operators execution order revisited (phantom overflow 
   paranoia)
  2.0_On 'add_liquidity' method added a var ("let pool_ratio = 
        pool1 / pool2") and some parentheses to 
        rearrange math operators execution order
  2.1_On 'internal_swap' methods added some parentheses to 
        rearrange math operators execution order
